data-prepper-plugins/opensearch/src/main/java/com/amazon/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -46,7 +47,7 @@
 import java.util.function.Supplier;
 
 @DataPrepperPlugin(name = "opensearch", pluginType = Sink.class)
-public class OpenSearchSink extends AbstractSink<Record<String>> {
+public class OpenSearchSink extends AbstractSink<Record<Object>> {
```

Comment:
IMO, we could scope the reference data type to Event since the only supported document format is json for opensearch sink. Is there any reason to use Object here?

Reply:
The sink. still needs to support trace pipelines which use `String`s. Leveraging an object here is only temporary. Once the trace plugins are updated to use the `Event` model we can convert this to and `Event` instead of an `Object`

Reply:
Onboard. Suggest adding a TODO comment just as you mentioned below.